agent/acs/handler/attach_eni_handler.go

```diff
@@ -119,13 +128,24 @@ func (attachENIHandler *attachENIHandler) handleSingleMessage(message *ecsacs.At
 func (handler *attachENIHandler) addENIAttachmentToState(message *ecsacs.AttachTaskNetworkInterfacesMessage) {
 	attachmentArn := aws.StringValue(message.ElasticNetworkInterfaces[0].AttachmentArn)
 	seelog.Info("Adding eni info to state, eni: %s", attachmentArn)
-	handler.state.AddENIAttachment(
-		&api.ENIAttachment{
-			TaskArn:          aws.StringValue(message.TaskArn),
-			AttachmentArn:    attachmentArn,
-			AttachStatusSent: false,
-			MacAddress:       aws.StringValue(message.ElasticNetworkInterfaces[0].MacAddress),
-		})
+	eniAttachment := &api.ENIAttachment{
+		TaskArn:          aws.StringValue(message.TaskArn),
+		AttachmentArn:    attachmentArn,
+		AttachStatusSent: false,
+		MacAddress:       aws.StringValue(message.ElasticNetworkInterfaces[0].MacAddress),
+	}
+	handler.state.AddENIAttachment(eniAttachment)
+
+	// Stop tracking the eni attachment after timeout
+	eniAckTimeout := time.Duration(aws.Int64Value(message.WaitTimeoutMs)) * time.Millisecond
+	timeoutHandler := func() {
```

Comment:
Can you make this a named method? Also, please pass in the mac address as a string parameter to this method rather than just implicitly retrieving it.

Also, please add a `Warn` log for this method.